### ISSUE TITLE
Impossible to run goss over ssh (no TTY forces goss to read config from STDIN)

### ISSUE THREAD

**Issue by taurus-forever** [aelsabbahy/goss#28]
Hi,

I have tried to integrate goss-based tests with Jenkins and noticed an issues using goss over SSH.

Problem: goss does stuck when you try to execute the test over the SSH. Reproducible in 100% cases.

STR: 
1) go to some server and generate new goss file: goss add port 22
```
root@spce:~# goss add port 22
Adding Port to './goss.json':

{
    "tcp:22": {
        "listening": true,
        "ip": "0.0.0.0"
    }
}
```
2) check it locally: 
```
root@spce:~# goss -g goss.json validate
..

Total Duration: 0.003s
Count: 2, Failed: 0
root@spce:~# 
```
3) go outside and try to execute the test over ssh:
```
me@host:~# ssh root@spce "goss -g goss.json validate"
```
Actual result: the process got stuck, no any output. 

I have compares strace output with normal goss call and call over ssh and noticed that goss is waiting for goss.json file from STDIN in case of call over SSH. See S_IFIFO on the picture:
![screenshot-2015-12-23-141450-area](https://cloud.githubusercontent.com/assets/1928266/11977180/a8faccb0-a980-11e5-9a14-4124ad489d65.png)

Workaround: specify ssh option -t to force pseudo-tty allocation:
```
me@host:~# ssh root@spce -t "goss -g goss.json validate"

..

Total Duration: 0.004s
Count: 2, Failed: 0
Connection to 127.0.0.1 closed.
```

I suppose it can/should be properly fixed on goss side. Tnx!

**Comment by stephanbuys** [comment]
We always use ssh, but I've checked and used STDIN by default so that I don't have to copy the goss.json files to the server. (`cat local_test.json | ssh root@foo 'sudo goss validate'`). 

I know its not a fix, but might help.

**Comment by taurus-forever** [comment]
Tnx for the hint about the workaround here, sure it will work.

While we are delivering all the tests using common Debian packages (to install all the necessary
dependencies), we prefer to wait for releasing the fix which has been committed already :+1: 

For the moment we force SSH to open TTY using "-t" option.

**Comment by aelsabbahy** [comment]
Thanks for the feedback on `goss`, I appreciate these bug reports as they help me make the tool better since I don't use it myself at my current/new job.

On a related note..  I'm personally very interested to see how a workflow with goss + [sup] (https://github.com/pressly/sup) works out and maybe adding a FAQ/wiki writeup on it. The two project names go perfect together *gossup*

Let me know if either of you have had any experience with this combination.

**Comment by taurus-forever** [comment]
I never used sup personally, we have puppet and ansible in place, but I will have a look.

P.S. for the history, if someone Googled workaround above...
JFYI, sometimes Jenkins cannot open TTY and produces output:
```
Pseudo-terminal will not be allocated because stdin is not a terminal.
```
You should double -t option in this case, like:
```
me@host:~# ssh root@spce -t -t "goss -g goss.json validate"
```

**Comment by aelsabbahy** [comment]
This is fixed in [v0.0.17](https://github.com/aelsabbahy/goss/releases/tag/v0.0.17)
